### ISSUE TITLE
Can't access interface properties

### ISSUE THREAD

**Issue by adirh3** [AvaloniaUI/Avalonia#11166]
**Describe the bug**
Accessing many properties of interfaces exposed by Avalonia are not possible since #11062 with no alternatives.

**To Reproduce**
Calling `window.PlatformImpl?.DesktopScaling` or any property of `IPlatformImpl` is not possible.
Basically any interface that has suffix `Impl` cannot be used anymore, even when exposed from a public property.

**Comment by rabbitism** [comment]
It is fixed in https://github.com/AvaloniaUI/Avalonia/pull/11124

**Comment by adirh3** [comment]
> It is fixed in #11124

Thanks that's a workaround specifically for `DesktopScaling`, but in general my issue was that all properties of any interface that's publicly exposed are now internal.

**Comment by maxkatz6** [comment]
@adirh3 that's not an issue, but expected for internal interfaces to be trully internal now. 
Please open issues for any API that you need, but is missing now. Some were already returned as part of stable API.

**Comment by adirh3** [comment]
> @adirh3 that's not an issue, but expected for internal interfaces to be trully internal now. Please open issues for any API that you need, but is missing now. Some were already returned as part of stable API.

I understand it, but I think this may be unintentional, since `IWindowBaseImpl`, `ITopLevelImpl`, are **public** interfaces and instances of them are exposed from `public` properties, e.g. `PlatformImpl` in `Window`.
Is this intended? It's super confusing IMO

**Comment by grokys** [comment]
> I think this may be unintentional, since IWindowBaseImpl, ITopLevelImpl, are public interfaces

Nope, that's the problem - they've always been considered by us to be [unstable](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Platform/IWindowBaseImpl.cs#L7) interfaces, but because we've never enforced any sort of access control on them, people have been using them as public interfaces without realizing (and these interfaces are [_old_](https://github.com/AvaloniaUI/Avalonia/commit/c999d45e94c3ba7b8f04ef973c17fe10be7ea9d1#diff-80e68fa2b84ee2433af8d53db9e6773d47618a5ac10c2efde841de68c0b5a2a2) and by this point incredibly hacked together). We have our hands tied when making changes to Avalonia because of this.

**Comment by adirh3** [comment]
I understand now that these changes were intentional, and I don't mind the change of hiding these interfaces. 
I think that if you should change these public properties to private as well, e.g. `PlatformImpl` as it's super confusing for users to have access to a property with instance that has zero accessible members.

**Comment by Tulesha** [comment]
Hello. 

In our application we need an opportunity to disable a window in a way like it is disabled when it has a modal child window.  
As I can see in a source code in [`Window.cs`](https://github.com/AvaloniaUI/Avalonia/blob/release/11.0.7/src/Avalonia.Controls/Window.cs) you invoke [`IWindowImpl.SetEnabled( bool isEnabled )`](https://github.com/AvaloniaUI/Avalonia/blob/b3a8ba25cf3e6f43fe89b8df91eb339bc361e482/src/Avalonia.Controls/Window.cs#L824). But unfortunately we can't invoke this as [`IWindowImpl`](https://github.com/AvaloniaUI/Avalonia/blob/release/11.0.7/src/Avalonia.Controls/Platform/IWindowImpl.cs) is **Unstable**.

Is there any way to make `IWindowImpl` (or only `SetEnabled` method) part of **Stable API**? 

Avalonia Version: `11.0.7`

**Comment by workgroupengineering** [comment]
Use IsEnabled="false" in your windows instance

**Comment by Tulesha** [comment]
> Use IsEnabled="false" in your windows instance

`InputElement.IsEnabled` is not what we need, because, even with `IsEnabled=false`, user can steel have access to press buttons in `WindowTitleBar` (he can also move Window and close window from the **TaskBar** ).

**Comment by timunie** [comment]
I think it is not really a bug here but by-design. If we want to debate about new API, better to open a new feature request or a discussion under 💡 ideas section.
